packages/processors/src/strategy/common/baseRecipientStatusUpdated.handler.ts

```diff
@@ -0,0 +1,112 @@
+import StatusesBitmap from "statuses-bitmap";
+import { getAddress } from "viem";
+
+import { Application, Changeset } from "@grants-stack-indexer/repository";
+import { ChainId, ProcessorEvent } from "@grants-stack-indexer/shared";
+
+import { ApplicationStatus, IEventHandler, ProcessorDependencies } from "../../internal.js";
+import { createStatusUpdate, isValidApplicationStatus } from "../helpers/index.js";
+
+type Dependencies = Pick<
+    ProcessorDependencies,
+    "logger" | "roundRepository" | "applicationRepository"
+>;
+
+type ApplicationUpdate = {
+    application: Application;
+    status: number;
```

Comment:
there's kinda a weird thing because the original implementation expected more statutes but currently only 3 of the status flow are in production. however the status on Index 0 is not one of them so I rather keep this as it is for now